src/app/cli/src/coda.ml

```diff
@@ -141,17 +141,76 @@ let daemon logger =
                  Deferred.return ll )
        in
        let%bind conf_dir =
-         if is_background then (
+         if is_background then
            let home = Core.Sys.home_directory () in
            let conf_dir = compute_conf_dir home in
+           Deferred.return conf_dir
+         else Sys.home_directory () >>| compute_conf_dir
+       in
+       let () =
+         if is_background then (
            Core.printf "Starting background coda daemon. (Log Dir: %s)\n%!"
              conf_dir ;
            Daemon.daemonize
              ~redirect_stdout:(`File_append (conf_dir ^/ "coda.log"))
              ~redirect_stderr:(`File_append (conf_dir ^/ "coda.log"))
-             () ;
-           Deferred.return conf_dir )
-         else Sys.home_directory () >>| compute_conf_dir
+             () )
+         else ()
+       in
+       (*Check if the config files are for the current version*)
+       let%bind () =
+         let del_files dir =
```

Comment:
I'm trying to figure out if it's okay if this uses Core file IO (and blocks) or if we need to use Async.

In a sense, given that this is in the critical path before Coda even starts up, maybe it's okay that this is synchronous? But really we want to avoid this sort of synchronous I/O where possible. It's even more important to do so when doing this sort of expensive operation (like removing lots of files).

Can you redo this with `Async.Sys` instead -- look at `Async.Sys.readdir`